src/app/containers/BulletedList/index.jsx

```diff
@@ -25,8 +44,15 @@ const BulletedListContainer = ({ blocks, className, ...rest }) => {
         script={script}
         service={service}
         dir={dir}
+        ref={linkBlock ? viewRef : null}
       >
-        <Blocks blocks={blocks} componentsToRender={componentsToRender} />
+        <Blocks
+          blocks={blocks.map(block => ({
+            ...block,
+            model: { ...block.model, clickTrackerRef },
```

Comment:
is there a better way to get the ref to the component rather than adding it to the data prop?

Reply:
The reason we went with this approach was because the bulletList/BulletListItem use the Block component (which is used in other places) so we didn't want to add another prop to that component. I think another option is forwardRef but this would also require (from our understanding) changing the Block component so we went with this approach instead.

Reply:
I wouldn't be hesitant to enhance the functionality of components as long as you don't breaking the existing functionality and API.

I might be proven wrong but it doesn't feel like the right thing to do to pass a click hander to a component through the Optimo data.

I'll check the branch out and have a play about and see if I can come up with another way.

Reply:
I think this works as expected https://github.com/bbc/simorgh/pull/9699 I used HOCs to enhance `componentToRender` components when click functionality is needed. Might be a better way 🤞

Reply:
Refactored and tested it. All looks fine. Thanks for looking into it.